zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -450,6 +450,1704 @@ pub fn zip_315_confirmations_test_steps<T: ShieldedPoolTester>(
     );
 }
 
+/// Tests max spendable funds within the given shielded pool in a
+/// single transaction.
+///
+/// The test:
+/// - Adds funds to the wallet in two notes with different confirmation heights
+/// - Checks that the wallet balances are correct.
+/// - Constructs a request to spend the whole balance to an external address in the
+///   same pool.
+/// - Builds the transaction.
+/// - Checks that the transaction was stored, and that the outputs are decryptable and
+///   have the expected details.
+pub fn spend_max_spendable_single_step_proposed_transfer<T: ShieldedPoolTester>(
+    dsf: impl DataStoreFactory,
+    cache: impl TestCache,
+) {
+    let mut st = TestBuilder::new()
+        .with_data_store_factory(dsf)
+        .with_block_cache(cache)
+        .with_account_from_sapling_activation(BlockHash([0; 32]))
+        .build();
+
+    let account = st.test_account().cloned().unwrap();
+    let dfvk = T::test_account_fvk(&st);
+
+    // Add funds to the wallet in a single note
+    let value = Zatoshis::const_from_u64(60000);
+    let (h, _, _) = st.generate_next_block(&dfvk, AddressType::DefaultExternal, value);
+
+    let confirmation_policy = ConfirmationsPolicy::new_symmetrical(
+        NonZeroU32::new(2).expect("2 is not zero"),
+        #[cfg(feature = "transparent-inputs")]
+        false,
+    );
+    st.generate_empty_block();
+    st.generate_empty_block();
+    st.generate_empty_block();
+
+    st.generate_next_block(&dfvk, AddressType::DefaultExternal, value);
+
+    st.scan_cached_blocks(h, 5);
+
+    // Spendable balance matches total balance
+    assert_eq!(
+        st.get_total_balance(account.id()),
+        Zatoshis::const_from_u64(120_000)
+    );
+    assert_eq!(
+        st.get_spendable_balance(account.id(), confirmation_policy),
+        value
+    );
```

Comment:
This check looks to me like it should fail; the confirmations policy requires two confirmations, but the second 60000 ZAT chunk was received in the block at the chain tip, which only has one confirmation.